compiler/visq/visq

```diff
@@ -0,0 +1,206 @@
+    parser.add_argument(
+        "-f",
+        "--fp32_circle",
+        type=str,
+        help="Path to the fp32 circle model.",
+        required=True)
+    parser.add_argument(
+        "-q",
+        "--q_circle",
+        type=str,
+        help="Path to the quantized circle model.",
+        required=True)
+    parser.add_argument(
+        "-d",
+        "--data",
+        type=str,
+        help=
+        "Path to the data used for inference. Random data will be used if this option is not given.",
+        required=False)
+    parser.add_argument(
+        "--mpeir_output",
+        type=str,
+        help="Path to the output dot file (qerror metric = mPEIR).",
+        required=False)
+    return parser
+
+
+def _verify_args(args):
+    """Verify the given arguments"""
+
+    valid_outputs = ['mpeir_output', 'mse_output', 'mpesr_output']
+
+    # Check if at least one output option is given
+    num_outputs = 0
+    for output_name in valid_outputs:
+        if valid_attr(args, output_name):
+            num_outputs += 1
+
+    if num_outputs == 0:
+        raise RuntimeError("At least one output should be given.")
+
+
+def _run_dalgona(model, data, analysis, save_dir):
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    dalgona_path = os.path.join(dir_path, 'dalgona')
+    cmd = [dalgona_path]
+    cmd += ['--input_model', str(model)]
+    cmd += ['--analysis', str(analysis)]
+    if data != None:
+        cmd += ['--input_data', str(data)]
+    cmd += ['--analysis_args', str(save_dir)]
+
+    try:
+        subprocess.run(cmd, capture_output=True, check=True)
+    except subprocess.CalledProcessError as e:
+        print('Error raised while running the below command')
+        print(' '.join(cmd))
+        print(e.output)
+        raise
+
+
+def _fake_quantize(input_model, output_model):
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    circle_quantizer_path = os.path.join(dir_path, 'circle-quantizer')
+    cmd = [circle_quantizer_path]
+    cmd += ['--fake_quantize']
+    cmd += [str(input_model)]
+    cmd += [str(output_model)]
+
+    try:
+        subprocess.run(cmd, capture_output=True, check=True)
+    except subprocess.CalledProcessError as e:
+        print('Error raised while running the below command')
+        print(' '.join(cmd))
+        print(e.output)
+        raise
+
+
+# Recursively visit items and check if there is Infinity or NaN
+def _check_float(item):
+    if isinstance(item, dict):
+        for v in item.values():
+            _check_float(v)
+    if isinstance(item, list):
+        for v in item:
+            _check_float(v)
+    if isinstance(item, float):
+        if item == -float('inf') or item == float('inf'):
+            raise RuntimeError('Infinite value detected. Value must be float')
+        if math.isnan(item):
+            raise RuntimeError('NaN value detected. Value must be float')
+
+
+def _build_json(**kwargs):
```

Comment:
I think if the number of arguments are not arbitrary, it would be better to use named arguments. Explicit is better than implicit. And, we'd be able to use type annotation.
```suggestion
def _build_json(model: str, metric: str, colorscheme: list[str], error: dict):
```

Reply:
(optional) maybe make this a class and use setters to set attributes and provide `build()` like method.